Lua/Lua.sublime-syntax

```diff
@@ -1,87 +1,464 @@
-    - match: '\+|-|%|#|\*|\/|\^|==?|~=|<=?|>=?|(?<!\.)\.{2}(?!\.)'
-      scope: keyword.operator.lua
+      pop: true
+
+  variable:
+    - match: '{{identifier}}{{function_call_ahead}}'
+      scope: variable.function.lua
+      pop: true
+
+    - match: '{{identifier}}'
+      scope: variable.other.lua
+      pop: true
+
+  number:
+    - match: 0[Xx]\h*(?:\.\h*)?([Pp][-+]?\d*)?
+      scope: constant.numeric.hexadecimal.lua
+      pop: true
+
+    - match: \d+(?:\.\d*)?([Ee][-+]?\d*)?
+      scope: constant.numeric.decimal.lua
+      pop: true
+
+  string:
+    - match: \'
+      scope: punctuation.definition.string.begin.lua
+      set:
+        - meta_scope: string.quoted.single.lua
+        - include: string-content
+        - match: \'
+          scope: punctuation.definition.string.end.lua
+          pop: true
+
+    - match: \"
+      scope: punctuation.definition.string.begin.lua
+      set:
+        - meta_scope: string.quoted.double.lua
+        - include: string-content
+        - match: \"
+          scope: punctuation.definition.string.end.lua
+          pop: true
+
+    - match: \[(=*)\[
+      scope: punctuation.definition.string.begin.lua
+      set:
+        - meta_scope: string.quoted.block.lua
+        - match: \]\1\]
+          scope: punctuation.definition.string.end.lua
+          pop: true
+
+  string-content:
+    - match: \\[abfnrtv\\'"\n]
+      scope: constant.character.escape.lua
+
+    - match: \\x\h\h
+      scope: constant.character.escape.hexadecimal.lua
+
+    - match: \\d\d{1,3}
+      scope: constant.character.escape.decimal.lua
+
+    - match: \\u\{\h+\}
+      scope: constant.character.escape.unicode.lua
+
+    - match: \\z
+      push:
+        - meta_scope: constant.character.escape.whitespace.lua
+        - include: else-pop
+
+    - match: \n
+      scope: invalid.illegal.newline.lua
+    - match: \\.
+      scope: invalid.illegal.invalid-escape.lua
+
+  parenthesized-expression:
+    - match: \(
+      scope: punctuation.section.group.begin.lua
+      set:
+        - meta_scope: meta.group.lua
+        - match: \)
+          scope: punctuation.section.group.end.lua
+          pop: true
+        - match: (?=\S)
+          push: expression
+
+  table-constructor:
+    - match: \{
+      scope: punctuation.section.block.begin.lua
+      set:
+        - meta_scope: meta.mapping.lua
+        - match: \}
+          scope: punctuation.section.block.end.lua
+          pop: true
+
+        - match: '[,;]'
+          scope: punctuation.separator.field.lua
+
+        - match: \[
+          scope: punctuation.section.brackets.begin.lua
+          push:
+            - meta_scope: meta.brackets.lua
+            - match: \]
+              scope: punctuation.section.brackets.end.lua
+              pop: true
+            - match: (?=\S)
+              push: expression
+
+        - match: '{{identifier}}(?=\s*=)'
+          scope: meta.key.lua
+
+        - match: =
+          scope: punctuation.separator.key-value.lua
```

Comment:
To me, it's no different from a colon in a JavaScript object or a Python dict or the arrow in a Ruby hash. I'd rather highlight it accordingly based on syntactic function than highlight it as an operator just because it's an equal sign.

if we get complaints, we can always change it (or encourage the complainers to update their color schemes).